AmazonPublisherServicesAdapter/src/main/java/com/chartboost/mediation/amazonpublisherservicesadapter/AmazonPublisherServicesAdapter.kt

```diff
@@ -391,93 +472,144 @@ class AmazonPublisherServicesAdapter : PartnerAdapter {
                 return@suspendCancellableCoroutine
             }
 
-            buildAdRequestSize(request.format, adRequest, isVideo, preBidSettings)
-            buildCcpaPrivacy(adRequest, ccpaPrivacyString)
+            CoroutineScope(Main.immediate).launch {
+                preBiddingListener?.onPreBid(
+                    context,
+                    AmazonPublisherServicesAdapterPreBidRequest(
+                        chartboostPlacement = request.chartboostPlacement,
+                        format = request.format,
+                        amazonSettings = amazonSettings,
+                        ccpaPrivacyString = ccpaPrivacyString,
+                    )
+                )?.fold({ adInfo ->
+                    CoroutineScope(Main.immediate).launch {
+                        placementToPreBidAdInfoMap[placement] = adInfo
+                    }
```

Comment:
We already are inside a Main scope, does `fold` change the scope? 🤔  I suppose, always good to be safe.